File: call-clj/src/main/java/capclug/demo/callclj/Two.java
/*
 * $Id$
 * Copyright (C) 2010 ARIN
 */
package capclug.demo.callclj;

import capclug.demo.cljfromjava.two.Foo;

/**
 * @version $Rev$
 */
public class Two
{
    public static void main( String[] args )
    {
        Foo foo = new Foo("bar string", "baz string");
        System.out.println( "foo.bar is " + foo.bar );
        System.out.println( "foo.baz is " + foo.baz );
    }
}

File: compile-clj-to-java/src/main/clojure/capclug/demo/cljfromjava/two.clj
(ns capclug.demo.cljfromjava.two)

(defrecord Foo [bar baz])

(deftype Bing [ham ton])

(defprotocol Blah "doc string"
  (add [a b] (+ a b))
  (sub [a b] (- a b)))

(deftype Bleh [#^int c]
  Blah
  (add [a b] (+ a b c))
  (sub [a b] (- a b c)))

; there is a lot more that you could do here... 
